src/py/flwr/server/workflow/default_workflows.py

```diff
@@ -0,0 +1,346 @@
+            evaluate_workflow = default_evaluate_workflow
+        self.fit_workflow: Workflow = fit_workflow
+        self.evaluate_workflow: Workflow = evaluate_workflow
+
+    def __call__(self, driver: Driver, context: Context) -> None:
+        """Execute the workflow."""
+        if not isinstance(context, LegacyContext):
+            raise TypeError(
+                f"Expect a LegacyContext, but get {type(context).__name__}."
+            )
+
+        # Start the thread updating nodes
+        thread, f_stop = start_update_client_manager_thread(
+            driver, context.client_manager
+        )
+
+        # Initialize parameters
+        default_init_params_workflow(driver, context)
+
+        # Run federated learning for num_rounds
+        log(INFO, "FL starting")
+        start_time = timeit.default_timer()
+        cfg = ConfigsRecord()
+        cfg[KEY_START_TIME] = start_time
+        context.state.configs_records[CONFIGS_RECORD_KEY] = cfg
+
+        for current_round in range(1, context.config.num_rounds + 1):
+            cfg[KEY_CURRENT_ROUND] = current_round
+
+            # Fit round
+            self.fit_workflow(driver, context)
+
+            # Centralized evaluation
+            default_centralized_evaluation_workflow(driver, context)
+
+            # Evaluate round
+            self.evaluate_workflow(driver, context)
+
+        # Bookkeeping
+        end_time = timeit.default_timer()
+        elapsed = end_time - start_time
+        log(INFO, "FL finished in %s", elapsed)
+
+        # Terminate the thread
+        f_stop.set()
+        del driver
+        thread.join()
+
+
+def default_init_params_workflow(driver: Driver, context: Context) -> None:
+    """Execute the default workflow for parameters initialization."""
+    if not isinstance(context, LegacyContext):
+        raise TypeError(f"Expect a LegacyContext, but get {type(context).__name__}.")
+
+    log(INFO, "Initializing global parameters")
+    parameters = context.strategy.initialize_parameters(
+        client_manager=context.client_manager
+    )
+    if parameters is not None:
+        log(INFO, "Using initial parameters provided by strategy")
+        paramsrecord = compat.parameters_to_parametersrecord(
+            parameters, keep_input=True
+        )
+    else:
+        # Get initial parameters from one of the clients
+        log(INFO, "Requesting initial parameters from one random client")
+        random_client = context.client_manager.sample(1)[0]
+        # Send GetParametersIns and get the response
+        content = compat.getparametersins_to_recordset(GetParametersIns({}))
+        messages = driver.send_and_receive(
+            [
+                driver.create_message(
+                    content=content,
+                    message_type=MESSAGE_TYPE_GET_PARAMETERS,
+                    dst_node_id=random_client.node_id,
+                    group_id="",
+                    ttl="",
+                )
+            ]
+        )
```

Comment:
Let's put `message = driver.create_message(...)` on a line before `driver.send_and_receive` to make it easier to read